jax/interpreters/partial_eval.py

```diff
@@ -1045,7 +1052,7 @@ def write(x: Atom, b: bool) -> None:
   new_jaxpr = Jaxpr((),
                     [v for v, b in zip(jaxpr.invars, used_inputs)   if b],
                     [v for v, b in zip(jaxpr.outvars, used_outputs) if b],
-                    new_eqns[::-1])
+                    new_eqns[::-1], False)
```

Comment:
I hardcoded this here based on an idea (from the "IO and RNG" effects doc draft) that linear jaxprs would be effect-free. That is, we wrote signatures like this:

```
io_jvp : (a -> {IO} b) -> (a, T a) -> {IO} (b, T b)
io_lin : (a -> {IO} b) -> a -> {IO} (b, T a -o T b)
io_vjp : (a -> {IO} b) -> a -> {IO} (b, T b -o T a)
```

However, while that's likely true for some effects (particularly the ones considered in that IO and RNG doc), it may not be true for all. In particular, we might want debug prints in the backward pass!

Given we're now focused on effects which we want to include in linear jaxprs, I will indeed remove this hardcoding!